configman/value_sources/for_modules.py

```diff
@@ -189,10 +196,40 @@ def local_to_str(a_thing):
 
 
 #==============================================================================
+class OrderableObj(object):
+    """Python3 can't sort non-string types implicitly.
+    """
+    def __init__(self, value):
+        if not isinstance(value, six.string_types):
+            value_str = to_str(value)
+        else:
+            value_str = value
+        self.value_str = value_str
+        self.value = value
+    def __lt__(self, other):
+        return (self.value_str < other.value_str)
+    def __repr__(self):
+        return self.value
+
+
+#==============================================================================
+class OrderableTuple(object):
+    """Python3 can't sort non-string types implicitly.
+    """
+    def __init__(self, value, index=1):
+        self.value = value
+        self.index = index
+    def __lt__(self, other):
+        return (self.value[self.index] < other.value[self.index])
+    def __repr__(self):
+        return self.value
```

Comment:
We should add tests for these two things. I don't think it needs to be spectacular, but it'd be good to know that they work as advertised and that we don't break them in the future.

Also, do we need to add a `functools.total_ordering` decorator to these classes? If not, why not?